### PR TITLE
Handle range on heterogeneously typed Dataset column

### PR DESCRIPTION
**Pull request by philippjfr**
Calling range on a heterogeneously typed column currently causes all kinds of issues (including errors) due to python 3 sorting semantics. While range should generally not be called on object or string dtype columns it currently is so this needs to work robustly for the time being. Therefore this PR uses the python2sort utility to do the sorting robustly in python3.

### CODE REVIEW

**Comment by jlstevens** [comment]
This is one semantic change in Python 3 that I find to be downright annoying. Anyway, the suggested fix makes sense.

File: holoviews/core/data/pandas.py [hunk 1]
```diff
@@ -291,6 +294,16 @@ def dframe(cls, columns, dimensions):
...
 
+    @classmethod
+    def array(cls, dataset, dimensions):
+        if not dimensions:
+            dimensions = dataset.dimensions(label='name')
+        else:
+            dimensions = [dataset.get_dimensions(d).name for d in dimensions]
+        inds = [dataset.data.columns.index(dim.name) for dim in dimensions]
+        return dataset.data.values[:, inds]
+
```
Comment (jlstevens) [inline holoviews/core/data/pandas.py]: I don't quite see how this relates to the python 2 vs 3 sorting issue. Is this method implementation an unrelated fix?

Comment (philippjfr) [reply holoviews/core/data/pandas.py]: Yes, accidentally pushed this. Will revert.

**Comment by philippjfr** [comment]
Going to need rebuilt test data, but let's get https://github.com/ioam/holoviews/pull/1978 merged first since that also requires new test data.

**Comment by jlstevens** [comment]
Happy to merge when the tests go green.
